File: client/src/types/Manifest/Handler.ts
import {Contact, EmergencyPhone} from './Contact';

export interface Handler {
  name: string;
  epaSiteId: string;
  mailingAddress: Address;
  siteAddress: Address;
  contact: Contact;
  emergencyPhone: EmergencyPhone;
  // paperSignatureInfo: PaperSignature
  // electronicSignatureInfo?: ElectronicSignature[]
  order: number;
  registered: boolean;
  modified: boolean;
  limitedEsign: boolean;
  canEsign: boolean;
  siteState: Locality;
  hasRegisteredEmanifestUser: boolean;
  gisPrimary: boolean;
}

interface Address {
  address1: string;
  city: string;
  country: Locality;
  state: Locality;
  streetNumber: string;
  zip: string;
}

export interface Locality {
  code: string;
  name?: string;
}

File: server/apps/trak/serializers/handler.py
from rest_framework import serializers

from apps.trak.models import Handler
from apps.trak.serializers import AddressSerializer

from .contact import ContactSerializer, EpaPhoneSerializer
from .trak import TrakBaseSerializer


class HandlerSerializer(TrakBaseSerializer):
    """
    Handler model serializer for JSON marshalling/unmarshalling
    """

    epaSiteId = serializers.CharField(
        source="epa_id",
    )
    siteType = serializers.CharField(
        source="site_type",
        allow_null=True,
        required=False,
    )
    modified = serializers.BooleanField(
        allow_null=True,
        default=False,
    )
    # name
    mailingAddress = AddressSerializer(
        source="mail_address",
    )
    siteAddress = AddressSerializer(
        source="site_address",
    )
    contact = ContactSerializer()
    emergencyPhone = EpaPhoneSerializer(
        source="emergency_phone",
        allow_null=True,
        default=None,
    )
    # paperSignatureInfo
    electronicSignatureInfo = serializers.JSONField(
        source="electronic_signatures_info",
        allow_null=True,
        default=None,
    )
    registered = serializers.BooleanField(
        allow_null=True,
        default=False,
    )
    limitedEsign = serializers.BooleanField(
        source="limited_esign",
        allow_null=True,
        default=False,
    )
    canEsign = serializers.BooleanField(
        source="can_esign",
        allow_null=True,
        default=False,
    )
    hasRegisteredEmanifestUser = serializers.BooleanField(
        source="registered_emanifest_user",
        allow_null=True,
        default=False,
    )
    gisPrimary = serializers.BooleanField(
        source="gis_primary",
        allow_null=True,
        default=False,
    )

    def create(self, validated_data):
        return Handler.objects.create_handler(**validated_data)

    class Meta:
        model = Handler
        fields = [
            "epaSiteId",
            "siteType",
            "modified",
            "name",
            "siteAddress",
            "mailingAddress",
            "contact",
            "emergencyPhone",
            "electronicSignatureInfo",
            "registered",
            "limitedEsign",
            "canEsign",
            "hasRegisteredEmanifestUser",
            "gisPrimary",
        ]
